src/librustc_lint/builtin.rs

```diff
@@ -742,6 +742,40 @@ impl LateLintPass for Deprecated {
 }
 
 declare_lint! {
+    DEPRECATED_ATTR,
+    Warn,
+    "detects use of deprecated attributes"
+}
+
+/// Checks for use of attributes which have been deprecated.
+#[derive(Clone)]
+pub struct DeprecatedAttr;
+
+impl LintPass for DeprecatedAttr {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(DEPRECATED_ATTR)
+    }
+}
+
+impl EarlyLintPass for DeprecatedAttr {
+    fn check_attribute(&mut self, cx: &EarlyContext, attr: &ast::Attribute) {
+        let name = &*attr.name();
+        for &(n, _, ref g) in feature_gate::KNOWN_ATTRIBUTES {
+            if n == name {
+                if let &feature_gate::AttributeGate::Gated(feature_gate::Stability::Deprecated,
+                                                           ref name,
+                                                           ..) = g {
+                    cx.span_lint(DEPRECATED,
+                                 attr.span,
+                                 &format!("use of deprecated attribute: {}", name));
```

Comment:
This follows the format for deprecated items. That probably should have back ticks too (added), but we don't have the tracking issue number anywhere so we can't link to it very easily.